src/actions/lockDomainMetadata.ts

```diff
@@ -0,0 +1,21 @@
+import { ethers } from "ethers";
+import { Registrar } from "../contracts/types";
+import { validateOwner, validateStatus } from "./helpers";
+
+export const lockDomainMetadata = async (
+  domainId: string,
+  lockStatus: boolean,
+  potentialOwner: string,
```

Comment:
You can avoid having `potentialOwner` passed to this function and instead access the signer's address by doing `await registrar.signer.getAddress();`.

I recommend you have the 3 actions do this call, and then pass the `potentialOwner` to the validate functions.